syntax = "proto3";

package cms;

service CacheManagementService{
  // A simple RPC.
  // Obtains the MessageResponse at a given position.
rpc ProcessJobEndedMessage(JobEndedMessage) returns (MessageResponse) {}
 rpc GetNextBatchForJob(GetNextBatchForJobMessage) returns (GetNextBatchForJobResponse) {}
 rpc RegisterNewTrainingJob(RegisterTrainingJobMessage) returns (RegisterTrainingJobResponse) {}
 rpc GetServerResponse(Message) returns (MessageResponse) {}
}

//Incoming Messages
message JobEndedMessage{
  int32  job_id = 1;
}

message GetNextBatchForJobMessage{
  int32  job_id = 1;
  float avg_training_speed =2;
  float prev_batch_dl_delay = 3;
}

message RegisterTrainingJobMessage{
  int32 job_id = 1;
  int32 batch_size = 2;
}

message Message{
 string message = 1;
 
}

//Outgoing responses
message GetNextBatchForJobResponse{
  string batch_id = 1;
  string batch_metadata = 2;
  bool isCached = 3;
 }
//Outgoing responses
message RegisterTrainingJobResponse{
  string message = 1;
  string dataset = 2;
  bool registered = 3;
  int32 batches_per_epoch = 4;
 }

 message MessageResponse{
  string message = 1;
  bool received = 2;
 }
/*
cd sdl-server/protos
python -m grpc_tools.protoc --proto_path=. ./cms.proto --python_out=.. --grpc_python_out=..


python -m grpc_tools.protoc --proto_path=. ./cms.proto --python_out=../misc --grpc_python_out=../misc

*/